client/my-sites/checkout/get-thank-you-page-url/index.ts

```diff
@@ -516,6 +516,16 @@ function getFallbackDestination( {
 		}
 	}
 
+	const marketplaceProductSlugs =
+		cart?.products
+			?.filter( ( product ) => product?.extra?.is_marketplace_product )
+			?.map( ( product ) => product?.extra?.plugin_slug ) || [];
+
+	if ( marketplaceProductSlugs.length > 0 ) {
+		debug( 'site with marketplace products' );
+		return `/marketplace/thank-you/${ marketplaceProductSlugs.join( ',' ) }/${ siteSlug }`;
```

Comment:
As described at the top of this file, could you add a test to cover this behavior change?